components/client/src/main/java/com/hotels/styx/client/healthcheck/OriginHealthStatusMonitor.java

```diff
@@ -37,10 +38,11 @@
          *
          * @param id                the backend service id
          * @param healthCheckConfig configuration of the health check
-         * @param supplier          of the actual function which decided if the origin is healthy or not
+         * @param healthCheckFunction          of the actual function which decided if the origin is healthy or not
+         * @param client TODO
          * @return the monitor of the origin
          */
-        OriginHealthStatusMonitor create(Id id, HealthCheckConfig healthCheckConfig, Supplier<OriginHealthCheckFunction> supplier);
+        OriginHealthStatusMonitor create(Id id, HealthCheckConfig healthCheckConfig, Supplier<OriginHealthCheckFunction> healthCheckFunction, HttpClient client);
```

Comment:
Since we have moved the HttpClient creation outside of healthCheckFunction(), do we still need to use a supplier for the latter? I'm guessing it was used to  lazily create the HttpClient.

Reply:
I tried removing the `Supplier`, but this caused tests to fail. I put it back since the change was out of the scope of this task.